frontend/ui/elements/common_settings_menu_table.lua

```diff
@@ -55,6 +55,21 @@ if Device:setDateTime() then
     common_settings.time = {
         text = _("Time and date"),
         sub_item_table = {
+          {
+            text = _("12-hour clock"),
+            keep_menu_open = true,
+            checked_func = function()
+                return G_reader_settings:isTrue("twelve_hour_clock")
+            end,
+            callback = function()
+                local twelve_hr = G_reader_settings:isTrue("twelve_hour_clock")
+                if twelve_hr then
+                    G_reader_settings:flipFalse("twelve_hour_clock")
+                else
+                    G_reader_settings:flipTrue("twelve_hour_clock")
+                end
+            end,
```

Comment:
You can make that one line:
https://github.com/koreader/koreader/blob/761f332d47bc9ca0fd87f9425b86a87cd4a6d12d/frontend/apps/reader/modules/readerwikipedia.lua#L280-L282
Same for `shorten_home_dir` below.